src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,134 @@
+            statement.setLong(1, id);
+
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                book = new Book();
+                id = resultSet.getObject(1, Long.class);
+                String title = resultSet.getString(2);
+                BigDecimal price = resultSet.getBigDecimal(3);
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find the book: " + e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM book";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement
+                    = connection.prepareStatement(query);
+
+            ResultSet resultSet = statement.executeQuery();
+
+            while (resultSet.next()) {
+                Book book = new Book();
+                Long id = resultSet.getObject(1, Long.class);
+                String title = resultSet.getString(2);
+                BigDecimal price = resultSet.getBigDecimal(3);
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+                books.add(book);
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find all books " + e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE book "
+                + "SET title = ?, price = ? "
+                + "WHERE id = ? ;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement
+                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new RuntimeException(
+                        "Expected to update at least 1 row, but inserted 0 rows");
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't update the book " + e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM BOOK WHERE id = ? ;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement
+                    = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            int affectedRows = statement.executeUpdate();
+            return affectedRows > 1;
```

Comment:
The `deleteById` method returns `affectedRows > 1`, which is incorrect. It should return `affectedRows > 0` to indicate that at least one row was deleted. The current condition will incorrectly return `false` if exactly one row is deleted, which is the expected behavior for deleting a single book by ID.